### DISCUSSION TITLE
Incorrect sign-off using Codespaces

### DISCUSSION THREAD

**Discussion by douglascamata** [community/community#20599] [Codespaces]
Currently Codespaces is setup with `GitHub` as user name and `noreply@github.com` as email. Codespaces should inherit the username and email from the user instead of these defaults.

It's a huge waste of time for people working DCO checks that require correct commit sign-off.

There is a post in the Github Community that's almost 2 years old asking for this to be fixed: https://github.community/t/incorrect-signer-when-using-git-commit-s/137012

**Comment by douglascamata** [comment]
According to a comment at #20925, looks like some env vars are now not set anymore. I can confirm that they were set in my Codespace and manually unsetting them solves the issue.

Will try to create a new Codespace later to see if the problem will be gone by default.

**Comment by jacobwoffenden** [comment]
This is still an issue. Unsetting as per https://github.com/orgs/community/discussions/20925 breaks GPG signing in Codespaces as well

**Reply by pl4nty** [reply]
I'm seeing similar behaviour and unsetting or changing breaks signing. might be a regression though, I think it was fine earlier this year

**Comment by bam80** [comment]
I still see this issue, but the problem is not always reproducible for me.

Initially I had correct Signed-off-by: field in my commits, but later I occasionally realized that changed to GitHub noreply@github.com just by itself.
I found that apparently comes from the two env vars, so unsetting them manually helps.

My assumption is maybe they are somehow injecting to the workspace environment, but that happens not always immediately but over time.

I would be appreciate for a temporary solution, how could I unset these vars automatically?
Thanks.  

PS:
the repos I tried:
https://github.com/openwrt/openwrt
https://github.com/openwrt/packages
